tasks/update_go.py

```diff
@@ -0,0 +1,171 @@
+import hashlib
+import re
+import sys
+from typing import Dict, Optional, Set, Tuple, Type
+
+import requests
+from invoke import Context, exceptions, task
+
+# a platform is an OS and an architecture
+Platform = Type[Tuple[str, str]]
+
+# list all platforms used in the repo
+# store the platform names in the format expected by Go
+# the variables in the Dockerfiles should be stored in uppercased variables
+# see https://go.dev/dl/ for the full list of platforms
+PLATFORMS: Set[Platform] = {
+    ("linux", "amd64"),
+    ("linux", "arm64"),
+    ("linux", "armv6l"),
+    ("windows", "amd64"),
+}
+
+# list of Dockerfiles where we want to replace Go version and sha variables
+# hardcode the number of expected matches so that we can warn if we get something else
+DOCKERFILES: Dict[str, int] = {
+    "./circleci/Dockerfile": 2,
+    "./deb-arm/Dockerfile": 3,
+    "./deb-x64/Dockerfile": 1,
+    "./rpm-arm64/Dockerfile": 2,
+    "./rpm-armhf/Dockerfile": 2,
+    "./rpm-x64/Dockerfile": 2,
+    "./suse-x64/Dockerfile": 2,
+    "./system-probe_arm64/Dockerfile": 2,
+    "./system-probe_x64/Dockerfile": 2,
+}
+
+
+# returns a map from a pattern to what it should be replaced with, for dockerfiles
```

Comment:
you could put this as the method's docstring, wdyt? (same comment applies for other methods)